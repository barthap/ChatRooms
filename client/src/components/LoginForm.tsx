import React from 'react';

import 'bootstrap/dist/css/bootstrap.min.css';

interface Props {
  onSubmit?: (nickname: string) => void;
  errorMessage?: string | null;
}

export default function LoginForm({ onSubmit, errorMessage }: Props = {}) {
  const [nickname, setNickname] = React.useState('');

  const handleSubmit: React.FormEventHandler<HTMLFormElement> = event => {
    event.preventDefault();
    onSubmit && onSubmit(nickname);
    console.log(nickname);
  };

  return (
    <form onSubmit={handleSubmit}>
      <h3>Sign In</h3>

      <div className="form-group row mb-3">
        <label htmlFor="nickname" className="col-sm-3 col-form-label">
          Nickname:
        </label>
        <div className="col-sm-9">
          <input
            type="text"
            className="form-control"
            placeholder="Enter your nickname"
            id="nickname"
            name="nickname"
            value={nickname}
            onChange={e => setNickname(e.target.value)}
          />
        </div>
      </div>

      <div className="d-flex justify-content-between">
        <p className="text-danger">{errorMessage && <small>{errorMessage}</small>}</p>

        <button type="submit" className="btn btn-primary">
          Enter chat
        </button>
      </div>
    </form>
  );
}
