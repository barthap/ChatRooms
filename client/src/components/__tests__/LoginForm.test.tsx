import { render, screen } from '@testing-library/react';
import React from 'react';

import LoginForm from '../LoginForm';

describe('<LoginForm />', () => {
  it('renders form snapshot without error message', () => {
    const loginForm = render(<LoginForm />);
    expect(loginForm).toMatchSnapshot();
  });
  it('displays error message', () => {
    const msg = 'Test error';
    render(<LoginForm errorMessage={msg} />);
    const msgElement = screen.getByText(msg);
    expect(msgElement).toBeInTheDocument();
  });
});
