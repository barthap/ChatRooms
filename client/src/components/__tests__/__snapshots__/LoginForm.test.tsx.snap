// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> renders form snapshot without error message 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <h3>
          Sign In
        </h3>
        <div
          class="form-group row mb-3"
        >
          <label
            class="col-sm-3 col-form-label"
            for="nickname"
          >
            Nickname:
          </label>
          <div
            class="col-sm-9"
          >
            <input
              class="form-control"
              id="nickname"
              minlength="3"
              name="nickname"
              placeholder="Enter your nickname"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="d-flex justify-content-between"
        >
          <p
            class="text-danger"
          />
          <button
            class="btn btn-primary"
            type="submit"
          >
            Enter chat
          </button>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <h3>
        Sign In
      </h3>
      <div
        class="form-group row mb-3"
      >
        <label
          class="col-sm-3 col-form-label"
          for="nickname"
        >
          Nickname:
        </label>
        <div
          class="col-sm-9"
        >
          <input
            class="form-control"
            id="nickname"
            minlength="3"
            name="nickname"
            placeholder="Enter your nickname"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="d-flex justify-content-between"
      >
        <p
          class="text-danger"
        />
        <button
          class="btn btn-primary"
          type="submit"
        >
          Enter chat
        </button>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
